queue/src/workers/generate-mails.ts

```diff
@@ -146,24 +147,33 @@ export default function createWorker() {
 
       const xlsxBuffer = Buffer.from(await workbook.xlsx.writeBuffer());
 
-      // Add files to an archive
       const archive = archiver('zip');
       archive.append(xlsxBuffer, { name: `${name}-destinataires.xlsx` });
       archive.append(finalPDF, { name: `${name}.pdf` });
-      const command = new PutObjectCommand({
-        Bucket: 'zerologementvacant',
-        Key: `${name}.zip`,
-        ContentLanguage: 'fr',
-        Body: Readable.from(archive),
-        ContentType: 'application/x-zip',
-        ACL: 'authenticated-read',
+      const upload = new Upload({
+        client: s3,
+        params: {
+          Bucket: config.s3.bucket,
+          Key: `${name}.zip`,
+          Body: Readable.from(archive),
+          ContentLanguage: 'fr',
+          ContentType: 'application/x-zip',
+          ACL: 'authenticated-read',
+        },
+      });
+      const results = await Promise.all([archive.finalize(), upload.done()]);
+      const objectKey = results[1].Key;
+
+      const command = new GetObjectCommand({
+        Bucket: config.s3.bucket,
+        Key: objectKey,
       });
-      await Promise.all([archive.finalize(), s3.send(command)]);
 
       const signedUrl = await getSignedUrl(s3, command, {
```

Comment:
C'est une astuce pour éviter un bug dans le client AWS. Quand je demande l'URL signée sur la commande d'upload il y a une erreur sur la region qui n'est pas récupérée:

`EndpointError: A region must be set when sending requests to S3`

Passer par une requête supplémentaire `GetObject` permet de contourner cette erreur.